### PR TITLE
[UnifyHeaders]Create an interface for SerializedCamera, to allow share drawers across Pipelines

### PR DESCRIPTION
**Pull request by alex-vazquez**
### Purpose of this PR
https://jira.unity3d.com/browse/XPIPELINE-98

In this PR we are adding the interface, and it's implementation on both pipelines

---
### Testing status
- Check with two inspectors, and see that everything is updated on both when edited on a single Inspector
- Check multiple edition of cameras

---
### Comments to reviewers
This will allow us create a base class with drawers targetting `ISerializedCamera` and be able to share the same methods from URP and HDRP Camera Editors

### CODE REVIEW

**Review by jenniferd-unity — APPROVED** [review]

**Comment by alex-vazquez** [comment]
@Unity-Technologies/gfx-qa-hdrp , @Unity-Technologies/gfx-qa-urp coud you do a quick smoke test editing cameras? Multiple edition will be good too. Thank you a lot.

**Review by RSlysz — APPROVED** [review]
lgtm

**Review by TomasKiniulis — APPROVED** [review]
Looks solid! Did not encounter any PR specific issues.

Checked editing single and multiple selected cameras with multiple locked/unlocked inspectors.
Made sure that changes are applied for correct cameras with
- AntiAliasing
- Dynamic Resolution
- Volume Layer Mask
- Custom Frame Settings
- Field Of View
- various random toggles and inputs

https://user-images.githubusercontent.com/50582134/110657932-ed06fd80-81c9-11eb-91ef-d723993d15df.mp4

**Comment by sebastienlagarde** [comment]
PR need to merge amster. @martint-unity can you review? thanks

**Review by martint-unity — APPROVED** [review]
LGTM but please have a URP QA look at this.

File: com.unity.render-pipelines.universal/Editor/Camera/UniversalRenderPipelineCameraEditor.cs [hunk 1]
```diff
@@ -497,7 +478,7 @@ void DrawStackSettings()
             if (m_StackSettingsFoldout.value)
             {
                 m_LayerList.DoLayoutList();
```
Comment (hdb-unity) [inline com.unity.render-pipelines.universal/Editor/Camera/UniversalRenderPipelineCameraEditor.cs:480]: Got exceptions in console from line 480.

Steps:
1. new URP template (change manifest.json)
2. set layout to default
3. right click "inspector" > add tab > inspector
4. select a camera
5. in one inspector, change camera Render Type = overlay
6. change camera Render Type = base

Rendering the Stack list on second inspector causes exceptions in console.

**Review by hdb-unity — APPROVED** [review]
Tested in URP:
- after commit with bug fix for 1322865
- multiple inspectors
- multi- and single-edit
- camera component UI (Render Type, Projection, Rendering, Environment, Output, Stack)
- built player (OSX)

**Pull request merged**